From 97cadc7023ba936642b0dd3ec1833cf4c0d55c57 Mon Sep 17 00:00:00 2001
From: Gurdal Oruklu <gurdal_oruklu@comcast.com>
Date: Mon, 9 Mar 2026 20:23:45 +0000
Subject: [PATCH] RDKEMW-15176: increased swap limit for containers

---
 bundle/lib/source/DobbySpecConfig.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/bundle/lib/source/DobbySpecConfig.cpp b/bundle/lib/source/DobbySpecConfig.cpp
index c2605483..d339dfd9 100644
--- a/bundle/lib/source/DobbySpecConfig.cpp
+++ b/bundle/lib/source/DobbySpecConfig.cpp
@@ -65,6 +65,8 @@ static const ctemplate::StaticTemplateString MEM_LIMIT =
 static const ctemplate::StaticTemplateString MEM_SWAP_LIMIT =
     STS_INIT(MEM_SWAP_LIMIT, "MEM_SWAP_LIMIT");
 
+static constexpr unsigned MEM_SWAP_LIMIT_EXTRA_BYTES = 200u * 1024u * 1024u;
+
 static const ctemplate::StaticTemplateString CPU_SHARES_ENABLED =
     STS_INIT(CPU_SHARES_ENABLED, "CPU_SHARES_ENABLED");
 static const ctemplate::StaticTemplateString CPU_SHARES_VALUE =
@@ -1276,7 +1278,7 @@ bool DobbySpecConfig::processMemLimit(const Json::Value& value,
     }
 
     dictionary->SetIntValue(MEM_LIMIT, memLimit);
-    dictionary->SetIntValue(MEM_SWAP_LIMIT, memLimit + (200 * 1024 * 1024));
+    dictionary->SetIntValue(MEM_SWAP_LIMIT, memLimit + MEM_SWAP_LIMIT_EXTRA_BYTES);
 
     return true;
 }
-- 
2.34.1

