diff --git a/daemon/process/dobby.service b/daemon/process/dobby.service
index 567d5358..28024902 100644
--- a/daemon/process/dobby.service
+++ b/daemon/process/dobby.service
@@ -34,8 +34,8 @@
 [Unit]
 Description=RDK Dobby (Container) daemon
 Requires=dbus.socket dbus.service
-Before=wpeframework.service
-After=dbus.socket dbus.service
+Before=wpeframework.service entos-appmanagers-rfc-controller.service
+After=dbus.socket dbus.service entos-appmanagers-rfc-controller.service
 
 [Service]
 Type=notify
