From: Yogaraj Murali <Yogaraj_Murali@comcast.com>
Date: Thu, 30 Jan 2025 21:24:05 +0530
Subject: [PATCH] RDKEMW-1230:  Adding dependency apparmor.service for dobby

Source: Comcast
Upstream-Status: Inappropriate [Comcast specific]

Signed-off-by: Yogaraj Murali <Yogaraj_Murali@comcast.com>
---

Index: git/daemon/process/dobby.service
===================================================================
--- git.orig/daemon/process/dobby.service
+++ git/daemon/process/dobby.service
@@ -35,7 +35,7 @@
 Description=RDK Dobby (Container) daemon
 Requires=dbus.socket dbus.service
 Before=wpeframework.service
-After=dbus.socket dbus.service
+After=dbus.socket dbus.service apparmor.service
 
 [Service]
 Type=notify
