From: Aravindan NarasimhapuramChakravarthy <aravindan_narasimhapuramchakravarthy@comcast.com>
Date: Mon, 06 Mar 2023 17:18:22 +0000
Subject: [PATCH] RDK-39807 : Fix dobby build error in kirkstone

Source: Comcast
Upstream-Status: Pending

Signed-off-by: Aravindan NarasimhapuramChakravarthy <aravindan_narasimhapuramchakravarthy@comcast.com>
---

Index: git/bundle/lib/source/DobbySpecConfig.cpp
===================================================================
--- git.orig/bundle/lib/source/DobbySpecConfig.cpp
+++ git/bundle/lib/source/DobbySpecConfig.cpp
@@ -24,7 +24,7 @@
 #include "DobbySpecConfig.h"
 #include "DobbyTemplate.h"
 #include "IDobbyUtils.h"
-
+#include <algorithm>
 #include <array>
 #include <atomic>
 #include <grp.h>
