From f8f06ee5bcab811c5fd6741d3dab8993de4df493 Mon Sep 17 00:00:00 2001
From: "Vivek.A" <Vivek_Arumugam@comcast.com>
Date: Sat, 22 Apr 2023 17:12:14 +0000
Subject: [PATCH] Send SIGHUP if compositor is terminated

---
 src/essos/renderer-backend.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/essos/renderer-backend.cpp b/src/essos/renderer-backend.cpp
index a995b1ce..fc353e7d 100644
--- a/src/essos/renderer-backend.cpp
+++ b/src/essos/renderer-backend.cpp
@@ -611,8 +611,13 @@ void EGLTarget::onTerminated()
     WARN_LOG("Terminated. Essos ctx = %p", essosCtx);
     stop();
 
+    if(getenv("CLIENT_IDENTIFIER") != nullptr) {
+        kill(getppid(), SIGHUP);
+        WARN_LOG("Sent SIGHUP to parent [%d]", getppid());
+    }
+
     g_timeout_add_seconds(1, [](gpointer) {
-        WARN_LOG("Exit");
+        WARN_LOG("Exit [%d]", getpid());
        _exit(1);
        return G_SOURCE_REMOVE;
        }, nullptr);
-- 
2.17.1

